tests/form_test.py

```diff
@@ -114,6 +114,40 @@ def test_init(self):
                 self.f.sys_vars["y"] == "boolean")
 
 
+def test_declare():
+    g = GRSpec()
+    g.declare('a')
+    assert g.sys_vars == dict(a='boolean'), g.sys_vars
+    assert g.env_vars == dict(), g.env_vars
+    g.declare('a', 'b')
+    assert g.sys_vars == dict(a='boolean', b='boolean'), g.sys_vars
+    assert g.env_vars == dict(), g.env_vars
+    g.declare('c', env=True)
+    assert g.sys_vars == dict(a='boolean', b='boolean'), g.sys_vars
+    assert g.env_vars == dict(c='boolean'), g.env_vars
+    # attempt to redeclare "c" as sys var
+    with nt.assert_raises(AssertionError):
+        g.declare('c')
+    g = GRSpec()
```

Comment:
in particular, I am thinking of the use-case where `test_declare` fails and an assertion error is presented with the corresponding line. How quickly can the program state at the point of error be determined?